From ea9ec1544b5b738893e02a46a08ee08932bacea1 Mon Sep 17 00:00:00 2001
From: Haoran Wang <elven.wang@nxp.com>
Date: Wed, 16 Mar 2022 03:05:28 +0800
Subject: [PATCH 2/2] Add Network Commissioning cluster setup for thermostat
 app Linux

For issue #15261 the Linux lighting app failed to be commission by
chip-tool pairing ble-wifi command.
Based on the message in #14751, the Linux platform intented to
support
WiFi network by default instead Thread.

This patch add Network Commissioning cluster setup for WiFi device for
thermostat-app.

Signed-off-by: Haoran Wang <elven.wang@nxp.com>
---
 examples/thermostat/linux/main.cpp | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/examples/thermostat/linux/main.cpp b/examples/thermostat/linux/main.cpp
index 68598a48a..e5d92f584 100644
--- a/examples/thermostat/linux/main.cpp
+++ b/examples/thermostat/linux/main.cpp
@@ -21,12 +21,21 @@
 #include <app-common/zap-generated/ids/Clusters.h>
 #include <app/CommandHandler.h>
 #include <app/clusters/identify-server/identify-server.h>
+#include <app/clusters/network-commissioning/network-commissioning.h>
 #include <app/util/af.h>
+#include <platform/Linux/NetworkCommissioningDriver.h>
 
 using namespace chip;
 using namespace chip::app;
 // using namespace chip::app::Clusters;
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
+namespace {
+DeviceLayer::NetworkCommissioning::LinuxWiFiDriver sLinuxWiFiDriver;
+Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(0, &sLinuxWiFiDriver);
+} // namespace
+#endif
+
 bool emberAfBasicClusterMfgSpecificPingCallback(chip::app::CommandHandler * commandObj)
 {
     emberAfSendDefaultResponse(emberAfCurrentCommand(), EMBER_ZCL_STATUS_SUCCESS);
@@ -87,7 +96,12 @@ static Identify gIdentify1 = {
     chip::EndpointId{ 1 }, OnIdentifyStart, OnIdentifyStop, EMBER_ZCL_IDENTIFY_IDENTIFY_TYPE_VISIBLE_LED, OnTriggerEffect,
 };
 
-void ApplicationInit() {}
+void ApplicationInit()
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
+    sWiFiNetworkCommissioningInstance.Init();
+#endif
+}
 
 int main(int argc, char * argv[])
 {
-- 
2.25.1

